editor/src/components/context-menu-wrapper.tsx

```diff
@@ -147,6 +143,15 @@ export const ContextMenu = <T,>({ dispatch, getData, id, items }: ContextMenuPro
     [getData, dispatch, isDisabled, isHidden],
   )
 
+  const { add, remove } = useContextMenuState()
```

Comment:
just an idea - you can return an `onVisibilityChange` function from the hook and use it directly

Reply:
I really prefer not to do that tbh, I know there is a lot of 'returning functions' in the codebase. Personally, my hot take is that it's probably unnecessary and is a lot more confusion to deal with.

Feel like this interface does a good job of decoupling state/data from the UI component.